Adding from command line user provided config file option
Patch by stenosophy.github.io

--- src/lxterminal.c
+++ src/lxterminal.c
@@ -44,6 +44,9 @@
 #include "preferences.h"
 #include "unixsocket.h"
 
+/* command line config hack, declared external in settings.h */
+gchar *cmdline_config = NULL;
+
 /* Utilities. */
 static void terminal_get_border(Term * term, GtkBorder * border);
 static void terminal_save_size(LXTerminal * terminal);
@@ -1457,6 +1460,29 @@ gboolean lxterminal_process_arguments(gi
             cmd_len ++;
         }
 
+	/* --config=<fname> */
+        else if (strncmp(argument, "--config=", 9) == 0)
+        {
+            g_strfreev(arguments->config);
+            g_shell_parse_argv(&argument[9], &cmd_len, &arguments->config, NULL);
+            cmdline_config = arguments->config[0];
+        }
+        else if ((strcmp, "--config") == 0 || (strcmp(argument, "-c") == 0))
+        {
+            if (arguments->config != NULL) g_strfreev(arguments->config);
+            cmd_len = 0;
+            arguments->config = g_malloc(argc * sizeof(gchar *));
+            while (argc > 1)
+            {
+                argc--;
+                argv_cursor++;
+                arguments->config[cmd_len++] = g_strdup(*argv_cursor);
+            }
+            arguments->config[cmd_len] = NULL;
+            cmdline_config = arguments->config[0];
+            cmd_len = 0;
+        }
+
         /* --geometry=<columns>x<rows> */
         else if (strncmp(argument, "--geometry=", 11) == 0)
         {
@@ -1528,7 +1554,7 @@ gboolean lxterminal_process_arguments(gi
         else {
             printf("%s\n", usage_display);
             return FALSE;
-    }
+        }
     }
     /* Handle --loginshell. */
     if (arguments->command != NULL && cmd_len <= 2) {
--- src/lxterminal.h
+++ src/lxterminal.h
@@ -79,6 +79,7 @@ typedef struct _term {
 typedef struct _command_arguments {
     char * executable;              /* Value of argv[0]; points into argument vector */
     gchar * * command;              /* Value of -e, --command; memory allocated by glib */
+    gchar * * config;               /* Value of -c, --config; memory allocated by glib */
     int geometry_bitmask;
     unsigned int geometry_columns;           /* Value of --geometry */
     unsigned int geometry_rows;
--- src/setting.c
+++ src/setting.c
@@ -333,10 +333,17 @@ Setting * load_setting()
     gchar * user_config_path = g_build_filename(dir, "lxterminal.conf", NULL);
     g_free(dir);
     gchar * system_config_path = g_strdup(PACKAGE_DATA_DIR "/lxterminal/lxterminal.conf");
-    gchar * config_path = user_config_path;
+/*    gchar * config_path = user_config_path; */
+    gchar * config_path = cmdline_config;
     
     gboolean need_save = FALSE;
 
+    if ( ! g_file_test(cmdline_config, G_FILE_TEST_EXISTS))
+    {
+        /* invalid config file provided, try user config! */
+        config_path = user_config_path;
+    }
+
     if ( ! g_file_test(user_config_path, G_FILE_TEST_EXISTS))
     {
         /* Load system-wide settings. */
--- src/setting.h
+++ src/setting.h
@@ -82,6 +82,9 @@
 #define ZOOM_RESET_ACCEL_DEF "<Primary><Shift>parenright"
 
 /* User preferences. */
+
+extern char *cmdline_config;    /* defined in lxterminal.c */
+
 typedef struct _setting {
 
     GKeyFile * keyfile;         /* Pointer to GKeyFile containing settings */
